manifests/init.pp

```diff
@@ -136,6 +136,12 @@
   Boolean                                    $use_ticket_cache                 = $zookeeper::params::use_ticket_cache,
   Boolean                                    $remove_host_principal            = $zookeeper::params::remove_host_principal,
   Boolean                                    $remove_realm_principal           = $zookeeper::params::remove_realm_principal,
+  # Quorum SASL /!\ Only works with ZK 3.4.10 or more recent; disabled by default /!\
+  Boolean                                    $quorum_auth_enable_sasl          = $zookeeper::quorum_auth_enable_sasl,
```

Comment:
You're creating a circular reference to the same variable. Either use `$zookeeper::params::quorum_auth_enable_sasl` or some sane default value.